shiny/render/_dataframe.py

```diff
@@ -0,0 +1,320 @@
+            If there are more rows than can fit in this space, the grid will scroll.
+        summary
+            If `True` (the default), shows a message like "Viewing rows 1 through 10 of 20"
+            below the grid when not all of the rows are being shown. If `False`, the message
+            is not displayed. You can also specify a string template to customize the
+            message, for example: `"Viendo filas {start} a {end} de {total}"`.
+        row_selection_mode
+            Use `"none"` to disable row selection, `"single"` to allow a single row to be
+            selected at a time, and `"multi-toggle"` to allow multiple rows to be selected
+            by clicking on them individually.
+
+        Returns
+        -------
+        :
+            An object suitable for being returned from a `@render.data_frame`-decorated
+            output function.
+
+        See Also
+        --------
+        ~shiny.ui.output_data_frame
+        ~shiny.render.data_frame
+        """
+        import pandas as pd
+
+        self.data: pd.DataFrame = cast(
+            pd.DataFrame,
+            cast_to_pandas(
+                data,
+                "The DataGrid() constructor didn't expect a 'data' argument of type",
+            ),
+        )
+
+        self.width = width
+        self.height = height
+        self.summary = summary
+        self.row_selection_mode = row_selection_mode
+
+    def to_payload(self) -> object:
+        res: dict[str, Any] = json.loads(
+            # {index: [index], columns: [columns], data: [values]}
+            self.data.to_json(  # pyright: ignore[reportUnknownMemberType]
+                None, orient="split"
+            )
+        )
+        res["options"] = dict(
+            width=self.width,
+            height=self.height,
+            summary=self.summary,
+            row_selection_mode=self.row_selection_mode,
+            style="grid",
+        )
+        return res
+
+
+class DataTable(AbstractTabularData):
+    def __init__(
+        self,
+        data: object,
+        *,
+        width: Union[str, float, None] = "fit-content",
+        height: Union[str, float, None] = "500px",
+        summary: Union[bool, str] = True,
+        row_selection_mode: Union[
+            Literal["none"], Literal["single"], Literal["multi-toggle"]
+        ] = "none",
+    ):
+        """
+        Holds the data and options for a ``shiny.render.data_frame`` output, for a
+        spreadsheet-like view.
+
+        Parameters
+        ----------
+        data
+            A pandas `DataFrame` object, or any object that has a `.to_pandas()` method
+            (e.g., a Polars data frame or Arrow table).
+        width
+            A width for the data grid to occupy, in CSS units. The default is `fit-content`,
+            which sets the grid's width according to its contents. Set this to `100%` to use
+            the maximum available horizontal space.
```

Comment:
A nit: I noticed these lines are longer than the 88 chars that black wants to use. Is there a particular reason for that? They can be reflowed with Alt-Q in VS Code.